### DISCUSSION TITLE
My API routes doesn't working....always 405 error

### DISCUSSION THREAD

**Discussion by kysclient** [vercel/next.js#52182] [App Router]
my local env perfectly work. but in vercel prod mode, it always found 405 error. please help me.

here's my client code

chat.tsx 
------------
` const response = await fetch(endpoint, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                signal: controller.signal,
                body,
            });`

and here's my server code.   

pages/api/aichat.ts
--------------------
`export default async function handler (req: Request): Promise<Response> { }`

and here's my server code.   

and here's my vercel.json.
-----------------------------

`{
  "version": 2,
  "routes": [
    {
      "src": "/api/aichat",
      "dest": "/api/aichat.ts"
    }
  ]
}
`

**Comment by icyJoseph** [comment]
Are all of the env vars correctly set in the Vercel Dashboard?

**Reply by kysclient** [reply]
> Are all of the env vars correctly set in the Vercel Dashboard?

I haven't registered it separately in the dashboard, but I use firebase, and the things written in the .env file work correctly

**Reply by icyJoseph** [reply]
Are you using the edge config?

```js
export const config = {
  runtime: 'edge',
}

// your handler goes here
```

**Reply by kysclient** [reply]
> Are you doing using the edge config?
> 
> 
> 
> ```js
> 
> export const config = {
> 
>   runtime: 'edge',
> 
> }
> 
> 
> 
> // your handler goes here
> 
> ```

Oh...yes!!!

**Reply by icyJoseph** [reply]
Did that do the trick? :) if so don't forget to mark as answered

**Reply by kysclient** [reply]
Whats mean edge do the trick?

**Reply by icyJoseph** [reply]
Nah I just thought that was missing. 

I'm not sure if the Vercel JSON config file is the problem :/

**Reply by kysclient** [reply]
I've tried everything I can, rewrite, routes, etc... Maybe I'm writing something wrong? Assuming /pages/api/chat.ts, can you give me an example?

**Reply by icyJoseph** [reply]
Why do you need the Vercel JSON file?

**Reply by kysclient** [reply]
It's just... there seems to be nothing wrong with my code, but it keeps failing, so I try to do something

**Reply by icyJoseph** [reply]
You are using Next.js right? with pages directory. I'd give it a go to put env vars in the Vercel dashboard.

**Reply by kysclient** [reply]
yeah. Im using nextjs. But i dont think env variable s problem. code related to environment variables works fine.
